doc/element_analyzers.rst

```diff
@@ -0,0 +1,207 @@
+Writing Element Analyzers
+=========================
+
+Element Analyzers are plugins for `rosdistro-reviewer` that analyze changes in a git repository and provide feedback. They are the primary mechanism for extending the functionality of `rosdistro-reviewer` with new checks.
+
+Each analyzer is responsible for a specific type of analysis, such as linting, style checking, or validating the structure of specific files.
+
+Creating a New Analyzer
+-----------------------
+
+To create a new Element Analyzer, you need to:
+
+1.  Create a new class that inherits from :py:class:`rosdistro_reviewer.element_analyzer.ElementAnalyzerExtensionPoint`.
+2.  Implement the :py:meth:`~.ElementAnalyzerExtensionPoint.analyze` in your class.
+3.  Register your class as a plugin in your `setup.cfg` file.
+
+The `analyze` method
+~~~~~~~~~~~~~~~~~~~~
```

Comment:
I notice there are three steps in the list above, but only two sections.

````suggestion
Create a new class
~~~~~~~~~~~~~~

Create a new class that inherits from ``ElementAnalyzerExtensionPoint``.

```python
from rosdistro_reviewer.element_analyzer import ElementAnalyzerExtensionPoint
 
class TodoChecker(ElementAnalyzerExtensionPoint):
    # ...
```
```

The `analyze` method
~~~~~~~~~~~~~~~~~~~~
````